### PR TITLE
V2: change order_details objects from classes to functions, add support for specific tools

### PR DESCRIPTION
**Pull request by jreiberkyle**
## Summary
1. Changes `order_details` objects from classes to functions that return dicts. This greatly decreases the amount of code required to achieve building up a request for creating an order.

2. Further, no longer imports the objects at the top level. This is to avoid naming conflicts with the new functions (this was less of a concern when the objects were classes)

e.g. naming conflict: avoid 
```python
from planet import product

product = product(['20170614_113217_3163208_RapidEye-5'], 'analytic', 'REOrthoTile') # uh oh!
```

less chance of a naming conflict: better?
```python
from planet import order_details

product = order_details.product(['20170614_113217_3163208_RapidEye-5'], 'analytic', 'REOrthoTile') # we are good!
```

3. Also adds support for all of the tools supported in the Orders API (e.g. `toar_tool()`, which creates the description for a TOAR tool from arguments)

## Usability impact:

Context - use of `order_details` in `OrdersClient`:
```python
>>> async def main():
...     async with Session() as sess:
...         cl = OrdersClient(sess)
...         order_id = await cl.create_order(order_detail)
...
>>> asyncio.run(main())

```

before - 
```python
>>> from planet import OrderDetails, Product, Tool
>>>
>>> image_ids = ['20170614_113217_3163208_RapidEye-5']
>>> order_detail = OrderDetails(
...     'test_order',
...     [Product(image_ids, 'analytic', 'REOrthoTile')],
...     [Tool('toar', {'scale_factor': 10000}),
...      Tool('reproject', {'projection': 'WGS84', kernel: 'cubic'}),
...      Tool('tile', {'origin_x: -180, 'origin_y: -90, 'pixel_size': 0.000027056277056,
...                         'name_template': 'C1232_30_30_{tilex:04d}_{tiley:04d}'})
... )
...
```

after - 
```python
>>> from planet.api.order_details import (
...     build_request, product, toar_tool, reproject_tool, tile_tool)
...
>>> products = [
...     product(['20170614_113217_3163208_RapidEye-5'], 'analytic', 'REOrthoTile')])
... ]
...
>>> tools = [
...     toar_tool(scale_factor=10000),
...     reproject_tool(projection='WSG84', kernel='cubic'),
...     tile_tool(1232, origin_x=-180, origin_y=-90,
...               pixel_size=0.000027056277056,
...               name_template='C1232_30_30_{tilex:04d}_{tiley:04d}')
... ]
...
>>> order_request = build_request(
...     'test_order', products, tools)
...

```
Addresses #254

### CODE REVIEW

File: planet/__init__.py [hunk 1]
```diff
@@ -14,10 +14,7 @@
 from .api.http import Session
 from .api.models import Order
 from .api.orders import OrdersClient
-from .api.order_details import (
-    OrderDetails, Product, Notifications, Delivery, AmazonS3Delivery,
-    AzureBlobStorageDelivery, GoogleCloudStorageDelivery,
-    GoogleEarthEngineDelivery, Tool, ClipTool)
+from .api import order_details
```
Comment (sarasafavi) [inline planet/__init__.py:17]: ah I love it.

File: planet/api/order_details.py [hunk 2]
```diff
@@ -13,722 +13,445 @@
...
+    >>> order_request = build_request(
+    ...     'test_order', products, tools)
     ...
 
     ```
 
-    or from a dict describing the order detail:
+    Parameters:
+        name: Name of the order.
+        products: Product(s) from the Data API to order.
```
Comment (sarasafavi) [inline planet/api/order_details.py:60]: this is an extremely minor comment, but would it be nicer to define `products` strictly in terms of item IDs & asset types? In which case we could avoid the possibly-confusing reference to Data API completely

File: planet/api/order_details.py [hunk 3]
```diff
@@ -13,722 +13,445 @@
...
+            coregistered.
+    '''
+    return _tool('coregister', {'anchor_item': anchor_item})
+
+
+def file_format_tool(file_format: str) -> dict:
+    '''Create the API spec representation of a file format tool.
+
+    Parameters:
+        file_format: The format of the tool output. Either 'COG' or 'PL_NITF'.
```
Comment (sarasafavi) [inline planet/api/order_details.py:373]: file format tool support - love to see it!

File: planet/scripts/cli.py [hunk 4]
```diff
@@ -286,14 +286,21 @@ async def create(ctx, name, ids, bundle, item_type, email, cloudconfig, clip,
...
-        product = planet.Product(ids, bundle, item_type)
+        product = planet.order_details.product(ids, bundle, item_type)
     except planet.specs.SpecificationException as e:
         raise click.BadParameter(e)
 
-    notifications = planet.Notifications(email=email) if email else None
+    if email:
+        notifications = planet.order_details.notifications(email=email)
+    else:
+        notifications = None
```
Comment (sarasafavi) [inline planet/scripts/cli.py:296]: nice readability edit

**Review by sarasafavi — APPROVED** [review]
thank you for your patience @jreiberkyle !

File: planet/api/order_details.py (hunk 2 above)
Comment (jreiberkyle) [reply planet/api/order_details.py:60]: i agree fully! the question is whether we want to follow the naming convention of the orders api

**Pull request merged**